### DISCUSSION TITLE
How to connect more Atorch to one ESP32

### DISCUSSION THREAD

**Discussion by RenPio2001** [syssi/esphome-atorch-dl24#52] [Ideas]
Is it possible to connect a second Atorch to one ESP32, any tips?
Thanks

**Comment by syssi** [comment]
I've prepared an example here: https://github.com/syssi/esphome-atorch-dl24/pull/53/files

**Reply by RenPio2001** [reply]
Good morning,
Two Atorch DT3010 work on one ESP32, only one Atorch worked on the first approach, but after slimming down the program to my needs, it works. Changing the parameter: throttle: 1s gave effect.
Now it works as I wanted in my Home Assistant
Thank you very much for help.

**Reply by syssi** [reply]
Decreasing the log level will free a lot of resources too! :-)

**Reply by RenPio2001** [reply]
Is there anything else that can be changed in the program to free up resources ?

[code.txt](https://github.com/syssi/esphome-atorch-dl24/files/11752819/code.txt)

**Reply by syssi** [reply]
Please reduce the `log_level` to `INFO`.

**Comment by syssi** [comment]
@RenPio2001 Off-topic: Could you provide a debug log of your setup? I would be happy about some RAW frames of your Atorch device.

**Reply by syssi** [reply]
I've extracted these notification payloads from your log:

```
0xFF, 0x55, 0x01, 0x02, 0x00, 0x0A, 0x10, 0x00, 0x00, 0x75, 0x00, 0x00, 0x0A, 0x00, 0x00, 0x68, 0x0E, 0x00, 0x00, 0x64, 0x00, 0x00, 0x00, 0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x00, 0x3C, 0x00, 0x00, 0x00, 0x00, 0x8C
0xFF, 0x55, 0x01, 0x02, 0x00, 0x0A, 0x13, 0x00, 0x00, 0x76, 0x00, 0x00, 0x0A, 0x00, 0x00, 0x68, 0x0E, 0x00, 0x00, 0x64, 0x00, 0x00, 0x00, 0x00, 0x00, 0x16, 0x00, 0x00, 0x00, 0x00, 0x3C, 0x00, 0x00, 0x00, 0x00, 0x88
```

Could you provide a product photo of your device? I've trouble to find the Atorch DT3010 on f.e. Aliexpress.

**Reply by syssi** [reply]
https://pl.aliexpress.com/item/32827325908.html

**Reply by RenPio2001** [reply]
My Atorch DT3010 has an internal temperature sensor like the one in the link

**Reply by syssi** [reply]
Unfortunately it looks like the external temperature is not part of the BLE payload. There is just a single temperature measurement available.
